### ISSUE TITLE
Todo list ?

### ISSUE THREAD

**Issue by gbrooker** [alisdairjsmyth/node-ambiclimate#1]
Hi Alisdair,
Lucinda at AmbiClimate just pointed me in your direction, I had been working on a homebridge plugin for Ambi climate too. I had approached from quite a different direction, adapting a precious netatmo plugin to work with AmbiClimate. I just got it functioning yesterday, but now I have seen your code, I prefer your approach. 

My implementation includes a function, to obtain the authorization code. I've refactored it, and prepared it for your review through a pull request. 

I'd be more than happy to collaborate with you on the project. Do you have a list of things you'd like to do, perhaps I can pick one or two, so we avoid working on the same things in parallel.
Cheers
Guy

**Comment by alisdairjsmyth** [comment]
Guy,

Happy to collaborate.  I was in the process of trying to resolve the oauth2 client, and frankly hadn't given too much thought past that point.  Any thoughts you have are welcome.

As you would have seen I have two distinct projects: 
- [node-ambiclimate](https://github.com/alisdairjsmyth/node-ambiclimate): a wrapper for the API and 
- [homebridge-ambiclimate](https://github.com/alisdairjsmyth/homebridge-ambiclimate): the homebridge plugin

(I am happy to collaborate on both)

At the moment the plugin exposes the current temperature and relative humidity and provides a mechanism for turning the AC on (in comfort mode) and off.

As an immediate action I will incorporate your pull-request and fix the current homebridge-ambiclimate implementation.

Alisdair

**Comment by gbrooker** [comment]
That's great. Next thing I'd propose to contribute is my attempt at a ThermostatService as a home bridge Platform. I'll need to refactor somewhat to match your much cleaner structure, so it will take a few days or so...

**Comment by gbrooker** [comment]
Thanks for the acknowledgement in the README :)

**Comment by gbrooker** [comment]
Ambi just added a /device/mode call to the api !

Details on the api doc page 
G

**Comment by alisdairjsmyth** [comment]
Added mode function - available in npm v2.1.0

**Comment by gbrooker** [comment]
I'm almost ready to commit my proposed changes to provide a ThermostatService. I ran in to a couple of issues with the ambi API (not matching docs). I've reported them, and hopefully will be able to finalise by next week. I also made a couple of feature requests which would help improve integration with HomeKit for a future version.

I decided not to attempt to integrate the Platform code just yet, and plan to move to that as a second step, once the Thermostat code is agreed.

**Comment by alisdairjsmyth** [comment]
I look forward to seeing what you have done.

One thing I noticed after you pointed me at the new Mode endpoint was that there was an inconsistency between the modes in the App and the API. e.g. turning a device off with the App is reported as Manual through the API.

**Comment by gbrooker** [comment]
Yes, the Off mode is not correctly reported, the Comfort mode returns Climate not Comfort as per docs and the Away upper and lower seem to have been swapped. I think they knocked up the interface quickly for us and didn't have a chance to test much. 

I wrote to them but I guess they are on vacation this week due to Chinese new year. 

I also requested visibility of the AC mode, the AC fan and louver state, and an event stream. I expect they'll be back to the office next week.

Cheers
Guy

**Comment by alisdairjsmyth** [comment]
Any feedback from the folks at Ambi Climate?

**Comment by gbrooker** [comment]
Hi Alisdair, 

apologies for the silence, I've been away on vacation for a couple of weeks then catching up at the office. 

Yes the bug in the Ambi API is now fixed, I've been working on the homebridge interface today and have added regular polling to keep it updated (until Ambi add an event api). I'm fairly happy with result, and once I have a chance to clean up the debugging code later in the week, I will upload to my repository for you to take a look.

**Comment by alisdairjsmyth** [comment]
Sounds good.

Is it your intention to submit a pull request on the [homebridge-ambiclimate project](https://github.com/alisdairjsmyth/homebridge-ambiclimate)?

**Comment by aaronfisher** [comment]
Hello. Is there an update on the new API working with smartthings? I'd love to integrate mine.

**Comment by alisdairjsmyth** [comment]
@aaronfisher - I'd love to integrate mine with SmartThings also, but have not identified a way to integrate with Ambi Climate's oauth2 implementation within SmartThings Groovy.

**Comment by alisdairjsmyth** [comment]
Copied from the current Kickstarter campaign for Ambitious 2.0 ...

2. If you help us reach HK$1,170,000 we'll get going on "Connected Everything" - various smart home integrations such as Homebridge & SmartThings. We know this is a hot topic and we'll make sure to run a special survey to find out the top integrations on your wishlist!

**Comment by aaronfisher** [comment]
I saw that. I hope that they are able to produce one but I imagine it is a few months away, at best.
